ossindex/ossindex.go

```diff
@@ -36,40 +36,50 @@ const defaultOssIndexURL = "https://ossindex.sonatype.org/api/v3/component-repor
 // MaxCoords is the maximum amount of coords to query OSS Index with at one time
 const MaxCoords = 128
 
-var (
-	ossIndexURL string
-	logLady     *logrus.Logger
-)
-
-var dbCache *cache.Cache
+// Server is a struct that holds the OSS Index options, logger and other properties related to
+// communicating with OSS Index
+type Server struct {
+	Options types.Options
+	logLady *logrus.Logger
+	agent   *useragent.Agent
+	dbCache *cache.Cache
+}
 
-func init() {
-	dbCache = &cache.Cache{
-		DBName: "nancy-cache",
-		TTL:    time.Now().Local().Add(time.Hour * 12),
-	}
+// NoCacheNoProblems deletes the local database directory.
+func (o *Server) NoCacheNoProblems() error {
+	return o.dbCache.RemoveCache()
 }
 
-func getOssIndexURL() string {
-	if ossIndexURL == "" {
-		ossIndexURL = defaultOssIndexURL
+// New is intended to be the way to obtain a ossindex instance, where you have control of the options
+func New(logger *logrus.Logger, options types.Options) *Server {
+	ua := useragent.New(logger, useragent.Options{ClientTool: options.Tool, Version: options.Version})
+	return &Server{
+		logLady: logger,
+		Options: options,
+		agent:   ua,
+		dbCache: cache.New(logger, cache.Options{
+			DBName: options.DBCacheName,
+			TTL:    options.TTL,
+		}),
 	}
-	return ossIndexURL
 }
 
-// RemoveCacheDirectory deletes the local database directory.
-func RemoveCacheDirectory(logger *logrus.Logger) error {
-	return dbCache.RemoveCache(logger)
+// Default is intended to be a way to obtain a ossindex instance, with rational defaults set
```

Comment:
‘an ossindex’

Reply:
Went with `a Server` since I changed the name of the struct!